src/world/Server/Script/CreatureAIScript.cpp

```diff
@@ -78,8 +80,18 @@ bool SummonList::hasEntry(uint32_t entry) const
     return false;
 }
 
+void SummonList::doAction(int32_t action, StorageType const& summons)
+{
+    for (uint64_t const& guid : _storage)
+    {
+        Creature* summon = _creature->getWorldMapCreature(guid);
+        if (summon && summon->GetScript())
+            summon->GetScript()->DoAction(action);
+    }
+}
+
 CreatureAIScript::CreatureAIScript(Creature* creature) : mScriptPhase(0), summons(creature), mCreatureTimerCount(0), mAIUpdateFrequency(defaultUpdateFrequency),
-isIdleEmoteEnabled(false), idleEmoteTimerId(0), idleEmoteTimeMin(0), idleEmoteTimeMax(0), _creature(creature), linkedCreatureAI(nullptr)
+isIdleEmoteEnabled(false), idleEmoteTimerId(0), idleEmoteTimeMin(0), idleEmoteTimeMax(0), _creature(creature), linkedCreatureAI(nullptr), mCreatureAIScheduler(new CreatureAIFunctionScheduler(this))
```

Comment:
If you use "new" there has to be a "delete", where do you clean the storage?

Reply:
make a shared_ptr out of it must have gone missing thank you